MozillaFoundation/foundation.mozilla.org
# Improve front-end build process

Related to #9556 #9707 #10083

Simplified front end build process. Also fixed a bunch of errors in previous build/watch process. For details, see comments left on [the "Files changed" tab](https://github.com/MozillaFoundation/foundation.mozilla.org/pull/10809/files).


# To QA: Part 1
Run `docker-compose up`, wait a bit, and visit different pages on the site to make sure pages are still looking normal. If PR has bug, then the pages will look very broken.

# To QA: Part 2
Make SCSS / Tailwind class changes and make sure new stylings are shown. I suggest setting background color to an strange HEX value so we can verify SCSS/Tailwind do get rebuilt upon code changes.

## Test by making changes in `.html` file:
For example, 
- Go to `network-api/networkapi/wagtailpages/templates/wagtailpages/homepage.html`, 
- Change line 32 to something like `<div class="container tw-bg-[#f34c40]">`
- Wait for Terminal prints out something like the following
```bash
foundationmozillaorg-watch-static-files-1  | Processing network-api/networkapi/temp/frontend/_css/buyers-guide.compiled.css...
foundationmozillaorg-watch-static-files-1  | Processing network-api/networkapi/temp/frontend/_css/donate.compiled.css...
foundationmozillaorg-watch-static-files-1  | Processing network-api/networkapi/temp/frontend/_css/main.compiled.css...
foundationmozillaorg-watch-static-files-1  | Finished network-api/networkapi/temp/frontend/_css/buyers-guide.compiled.css in 11 s
foundationmozillaorg-watch-static-files-1  | Finished network-api/networkapi/temp/frontend/_css/main.compiled.css in 11 s
foundationmozillaorg-watch-static-files-1  | Finished network-api/networkapi/temp/frontend/_css/donate.compiled.css in 11 s
```
- Visit homepage locally
- Verify the background you set shows up

## Test by making changes in `.jsx` file:
For example,
- Go to `source/js/components/join/join.jsx`
- Change line 291 to `let headingClasses = classNames("tw-bg-[#31a244]",` (note: remember to use a new strange HEX)
- Wait for Terminal prints out the "processing/finished" logs
- Refresh the page
- Scroll to the bottom of the page and see if the background colour you set shows up in the newsletter signup section. e.g., notice the green background
<img width="1150" alt="image" src="https://github.com/MozillaFoundation/foundation.mozilla.org/assets/2896608/d3cf4a2d-ec95-4e02-a282-2ff1597e0cc1">


## Test by making changes in `.scss` file:
**It will take a lot longer for changes in `.scss` file to show because compiling SCSS into CSS is a slowwwwww process. Once that compilation is done, we then need to wait for`postcss` script to run and finish.**

To test,
- Go to `source/sass/main.scss`
- Add new styling code to the bottom of the page, e.g, 
```scss
#view-home .areas-of-focus {
  @apply tw-bg-[#e2d036];

  h2 {
    background-color: #6530ee;
  }
}
```
- Wait for Terminal prints out the "processing/finished" logs. This time multiple sets of them will be printed because any SCSS changes will trigger *ALL* of the `build:css:file:**` tasks. Wait until the log stops at `Waiting for file changes...` and stops outputting new log
- Refresh the page
- Verify new styling are reflected.


# To QA: Part 3

Check logs in Terminal. Errors noted in #9707 #10083 should no longer be there.
